From e5e6cfd7fa7827f286348b870986dd82d79873a5 Mon Sep 17 00:00:00 2001
From: jingpeiyang <jingpeiyang@eswincomputing.com>
Date: Thu, 10 Aug 2023 14:11:47 +0800
Subject: [PATCH] [build] Add riscv_use_rvv condition for riscv32

Because this commit da7ce06226fda8175f1a443cfca54be9b774016e has been
merged, so modify this todo.

Signed-off-by: jingpeiyang <jingpeiyang@eswincomputing.com>
Change-Id: I4140124948037c6e779cea57b4c8f11fc0baa5eb
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4769016
Reviewed-by: Clemens Backes <clemensb@chromium.org>
Commit-Queue: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#89484}
---
 BUILD.gn | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index ad9a5281e13..84fb2b59210 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1359,10 +1359,9 @@ config("toolchain") {
     defines += [ "__riscv_xlen=32" ]
     defines += [ "CAN_USE_FPU_INSTRUCTIONS" ]
 
-    # TODO(riscv32): Add condition riscv_use_rvv and riscv_rvv_vlen here after
-    # 4538202 merge.
-    if (target_is_simulator) {
+    if (riscv_use_rvv || target_is_simulator) {
       defines += [ "CAN_USE_RVV_INSTRUCTIONS" ]
+      defines += [ "RVV_VLEN=${riscv_rvv_vlen}" ]
     }
   }
 
-- 
2.35.1

